### PR TITLE
Output json

### PR DESCRIPTION
**Pull request by jbuck**
This is a first pass at JSON output; it was faster to write this patch than to wait for heka-cat :)

### CODE REVIEW

**Comment by jbuck** [comment]
One critical fail in this patch - it doesn't escape `"` in strings with `\"` so this completely breaks the JSON output. Need to add that too

**Comment by trink** [comment]
You will need more escaping than just the quote and depending on the character encoding you want to support (e.g UTF-8) some additional special handling on the binary data output.

**Comment by jbuck** [comment]
Okay, I've got some escaping in there now but I need to actually use that everywhere I'm printing strings.

Could you take a look at this patch and let me know what needs improvement?

File: src/cli/lsb_heka_cat.c [hunk 1]
```diff
@@ -59,6 +59,128 @@ read_string(int wiretype, const char *p, const char *e, lsb_const_string *s)
...
+      case '\f': fputs("\\f", stdout); break;
+      case '\n': fputs("\\n", stdout); break;
+      case '\r': fputs("\\r", stdout); break;
+      case '\t': fputs("\\t", stdout); break;
+      default:
+        if ('\x00' <= c[i] && c[i] <= '\x1f') {
+          fprintf(stdout, "\\u%04x", c[i]);
+        } else {
+          fputc(c[i], stdout);
+        }
```
Comment (trink) [inline src/cli/lsb_heka_cat.c]: Add a break after the default branch

File: src/cli/lsb_heka_cat.c [hunk 2]
```diff
@@ -59,6 +59,128 @@ read_string(int wiretype, const char *p, const char *e, lsb_const_string *s)
...
 }
 
+// https://stackoverflow.com/questions/7724448/simple-json-string-escape-for-c/33799784#33799784
+static void output_json_string(lsb_const_string cs)
+{
+  const char *c = cs.s;
+  for (int i = 0; i < (int)cs.len; i++) {
+    switch (c[i]) {
+      case '"': fputs("\\\"", stdout); break;
+      case '\\': fputs("\\\\", stdout); break;
```
Comment (trink) [inline src/cli/lsb_heka_cat.c]: The forward slash should be included here too

**Pull request closed without merging**